src/ORM/Rule/IsNotReferencedBy.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Cake\ORM\Rule;
+
+use Cake\Datasource\EntityInterface;
+use Cake\ORM\Association;
+use RuntimeException;
+
+/**
+ * Checks that given entity is not referenced by any row in the given table
+ */
+class IsNotReferencedBy
+{
+    /**
+     * The repository that will be checked
+     *
+     * @var array
+     */
+    protected $_repository;
+
+    /**
+     * Constructor.
+     *
+     * @param string $repository The association name for the repository which will be checked.
+     */
+    public function __construct($repository)
```

Comment:
Core [rules support](https://github.com/cakephp/cakephp/blob/master/src/ORM/Rule/ExistsIn.php#L80-L94) `Association` and `Table` objects as well in such cases, not only association name strings.